src/chat/__tests__/narrowsSelectors-test.js

```diff
@@ -34,6 +34,7 @@ describe('getMessagesForNarrow', () => {
       },
       messages,
       outbox: [],
+      realm: eg.realmState({ email: eg.selfUser.email }),
```

Comment:
It looks like this is setting `realm.email` in the state because that value gets passed directly as `ownEmail` to `isMessageInNarrow`, having been selected by `getOwnEmail` (that `getOwnEmail` use having been added in this same commit).

I do notice that `getOwnEmail` has this in its JSDoc:

```
 * Prefer using `getOwnUserId` or `getOwnUser`; see #3764.
```

(#3764)

It looks like `getOwnUser` itself uses `getOwnEmail`, so it's not obviously better, I guess.

But I could imagine that being changed someday; I could see `getOwnUser` instead using, e.g., `getOwnUserId` (there's even a comment on `getOwnUserId` suggesting to do so) and `getUserForId`.

If such a change were made, it would no longer be useful to just set `state.realm.email`; we'd want to set `state.realm.user_id`. Possibly that's totally fine, and a bridge we'll happily cross when we get to it. But I wondered if we might like to consider providing a simple way to grab `state.realm` from the example data such that it contains all the stuff that's specific to the self-user: there's `email`, `user_id`, probably more.

Reply:
> But I wondered if we might like to consider providing a simple way to grab state.realm from the example data such that it contains all the stuff that's specific to the self-user: there's email, user_id, probably more.

Yeah, I agree -- that would be useful, and we'll want to add something like that. A more "batteries included" example Redux state, one might call it.

The reason I haven't yet gone and done that is that there are some API choices to be made about it:
* Probably it includes the self-user's email and related properties, as given by `eg.selfUser`, in `state.realm`.
* Probably it also includes `eg.selfUser` in `state.users`, and a corresponding account (with `eg.realm`) in `state.accounts`.
* Does it include `eg.otherUser` in `state.users`? Also `eg.thirdUser`? `eg.crossRealmBot`?
* There's `eg.stream` and `eg.subscription`. Does it include those in `state.streams` and `state.subscriptions`?
* Maybe there's even several different levels of "batteries included" that we want? In addition to probably still the initial state, the thing that's now `eg.reduxState`.
* What about when we add new example values along those lines, like the recent `thirdUser`?
  * If they're included, that potentially changes the behavior of existing tests. If it makes them fail, that's one thing; more insidiously, it could potentially make them pass trivially, even if the code they're meant to test stops behaving the way they're meant to check that it does.
  * If they're left out, then the "batteries included" state comes to include only some batteries, and potentially a less and less predictable subset of them.
  * Probably the right answer is to include new example values, and make a practice of systematically writing tests so that they don't depend on any assumptions that the state doesn't have miscellaneous other data they weren't expecting. That is, if a test makes its own `eg.makeStream()` call, it can assume *that* stream isn't in the example state; but it can't make an assumption like "there's only one stream in the state".
    * This requires a bit of work finding a clear pattern to prescribe for how to systematically do that.

This diff (as you saw, there are a good handful of lines like this one) definitely made me think about adding this. :slightly_smiling_face: Not enough that I felt I wanted to do it before shipping this branch, but it moved that feeling closer.

Reply:
> Not enough that I felt I wanted to do it before shipping this branch, but it moved that feeling closer.

Makes sense, that's fine! So I'd say you can merge this branch at will.